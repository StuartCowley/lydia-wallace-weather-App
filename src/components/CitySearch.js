import React from "react";
import PropTypes from "prop-types";

function CitySearch({ searchTerm, getForecast, setSearchTerm, setFirstLine }) {
  const handleChange = (event) => {
    setSearchTerm(event.target.value);
  };
  const handleSubmit = (event) => {
    event.preventDefault();
    getForecast(
      `https://mcr-codes-weather-app-alt.herokuapp.com/forecast?city=${searchTerm}`
    );
    setSearchTerm("");
    setFirstLine("");
  };

  return (
    <div className="city-search">
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          placeholder="Search for a city"
          onChange={handleChange}
          value={searchTerm}
        />
        <button type="submit" onClick={handleSubmit}>
          Submit
        </button>
      </form>
    </div>
  );
}

CitySearch.propTypes = {
  getForecast: PropTypes.func.isRequired,
  setSearchTerm: PropTypes.func.isRequired,
  searchTerm: PropTypes.string.isRequired,
  setFirstLine: PropTypes.func.isRequired,
};

export default CitySearch;
