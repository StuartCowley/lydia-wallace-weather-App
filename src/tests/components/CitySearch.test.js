import React from "react";
import { render } from "@testing-library/react";
import CitySearch from "../../components/CitySearch";

describe("CitySearch", () => {
  const validProps = {
    searchTerm: "",
    setSearchTerm: () => {},
    getForecast: () => {},
    setFirstLine: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <CitySearch
        searchTerm={validProps.searchTerm}
        getForecast={validProps.getForecast}
        setSearchTerm={validProps.setSearchTerm}
        setFirstLine={validProps.setFirstLine}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // it("renders correct placeholder text", () => {});
  // it("renders a button with 'submit'", () => {});
  // it("updates the data being displayed", () => {});
  // it("updates the first line based on the response code", () => {});
});
