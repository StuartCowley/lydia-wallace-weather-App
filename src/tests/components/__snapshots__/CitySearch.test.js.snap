// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitySearch renders correctly 1`] = `
<DocumentFragment>
  <div
    class="city-search"
  >
    <form>
      <input
        placeholder="Search for a city"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
